calico/felix/refcount.py

```diff
@@ -97,6 +97,14 @@ def on_object_startup_complete(self, object_id, obj):
             _log.info("Ignoring on_object_startup_complete for instance "
                       "in state %s", obj.ref_mgmt_state)
             return
+            # MD4 It's not obvious to me why this branch is needed - I suspect it's because of asynchronous completion
+            # (thanks to actors) of startup while the manager thinks it's being (e.g.) destroyed, but this definitely
+            # needs commenting at least.
+
+            # MD4 On a related note, it wasn't clear to me what the intended way of
+            # covering these sort of things is.  The use of call_via_cb in the test code is
+            # very opaque - is there a reason you used these rather than something equivalent
+            # to the N-BASE's TB_SCHEDULER / SCHEDULE_ONE()?
```

Comment:
The first iteration of the code used AsyncResult as its primary means of dealing with result passing.  I think it's an artifact of that.  I don't think I can justify rewriting the UTs to use a different technique but thanks for the feedback nonetheless.